docs/reference/monitoring/troubleshooting.asciidoc

```diff
@@ -27,3 +30,14 @@ monitoring data by using {metricbeat} the indices have `-mb` in their names. If
 the indices do not exist, review your configuration. For example, see
 <<monitoring-production>>.
 
+[[monitoring-troubleshooting-uuid]]
+=== Some monitoring data is missing from {kib}
+
+*Symptoms*:
+The information about some nodes, instances, or Beats in your cluster are
+missing from the *Stack Monitoring* page in {kib}.
+
+*Resolution*:
+Verify that the missing items have unique UUIDs. Each {es} node, {ls} node,
+{kib} instance, and Beat is considered unique based on its persistent UUID,
+which is found in its `path.data` directory.
```

Comment:
For Beats, I'd say that the way to ensure that each Beat instance (or whatever name @dedemorton suggests per the other comment) is seen as unique in the Stack Monitoring UI, the Beat's `path.data` folder must be non-existent or empty after installation.

@jakelandis I imagine an ES node's UUID is stored somewhere in it's data folder? Is it sufficient to say (from a user's perspective) that for a node to be given a new UUID it's data folder would have to not exist or be empty - same as the statement I'm making above for Beats?

@tylersmalley Same question as ES above, but for Kibana. I happen to know that the UUID is stored in `data/uuid` but, in general, we'd want a new Kibana instance to have an empty or non-existent data folder, right?

@jsvd Same question as ES above, but for Logstash. I happen to know that the UUID is stored in `data/uuid` but, in general, we'd want a new Logstash node to have an empty or non-existent data folder, right?

Reply:
Thanks @ycombinator I agree this will be useful information to add for all of the products.

Reply:
@ycombinator - We have a couple ways to reference the node, all of is stored in cluster state. Cluster state is technically stored in the data dir, but not a readable format. i.e. If you delete the data folder, you will get new a node id. The IDs are in the logs (at startup) and would suggest to reference the logs to get ids needed for troubleshooting.

Reply:
@ycombinator that is correct.

Reply:
@ycombinator Logstash will check if the `uuid` file exists in `path.data`, and create it otherwise. For running instances one can also retrieve the id through LS' API (i.e. `curl -s localhost:9600 | jq ".id"`)

Reply:
Maybe a followup step here would be to explain how to check if a uuid is showing up at all using a query in Elasticsearch. If the uuid doesn't show up at all then data isn't being generated or reaching the cluster, if it exists maybe we could do a second query to list ips associated with that uuid?

Reply:
Thanks @jakelandis, @tylersmalley, and @jsvd for your inputs about ES, Kibana, and LS respectively.

So the context for this discussion is that we often see users copying over our product's install folders, including the data path. This leads to two instances/nodes having the same UUID and, as a result, users see fewer instances/nodes in the Stack Monitoring UI than they are expecting. So we're trying to add docs here about how to troubleshoot this issue.

> if it exists maybe we could do a second query to list ips associated with that uuid?
Yes! In fact, we could have a Cluster Alert for this situation. Just created https://github.com/elastic/kibana/issues/64779 to track this feature. Let's continue any related discussion there. Thanks @jsvd!